src/layers/icmp.rs

```diff
@@ -134,6 +197,36 @@ impl Layer for ICMP {
                 })
             }
         };
+
+
+        self.data = match self.icmp_type {
+            ICMP_ECHO_REPLY | ICMP_ECHO_REQUEST => {
+                decoded = bytes.len();
+                IcmpData::Unknown(IcmpUnknownData { data: bytes[8..].to_vec() })
```

Comment:
This is more of a question. Can you please point to the relevant section in the RFC for me to better understand what you are trying to do?

Reply:
![image](https://github.com/ystero-dev/scalpel/assets/96462743/4c041130-1a81-4367-a80f-255d871d4198)
Here, the data part in ICMP Echo Request and Reply are unknown and variable. The reply just sends back the same data it gets in the request.

Reply:
Better to call it just `Data` than `UnknownData`. That sounds a bit confusing.